### ISSUE TITLE
ames: ack crashed %exit after removing a naxplanation flow on closing bones

### ISSUE THREAD

**Issue by yosoyubik** [urbit/urbit#6451]
[Full output](https://gist.github.com/yosoyubik/2fea18cc270b418b35777131b94e883d)

A publisher is, after sending an ack for a %cork, removing the associated flow—and in this case, it had previously also sent nacks (e.g. if it crashed when it received the %watch), so we remove the naxplanation bone as well.

We tell %behn to reset the timer for the backward bone to stop sending %boons, but we don't do that for the naxplanation bone—and that's the one we crashed on, so it looks like we could be trying to send that again, and when we wake up, the flow is gone, so we crash here `?<  =(~ next-wake.state)` because we are doing *message-pump-state.

We have [a check](https://github.com/urbit/urbit/blob/develop/pkg/arvo/sys/vane/ames.hoon#L2486) to avoid this kind of thing happening. If we wake up and need to sent some timed out packets, but the bone is corked, we no-op, but... we are not doing that for naxplanations that use a  different bone `(mix 0b10 bone)`, so we should probably be adding that bone to the check.

Now, this error shouldn't repeat for the same ship—and this is what I can see on ~norsyr-torryn. @zalberico, after doing `|ames-verb %odd`, do you see a "remove naxplanation flow" message?

**Comment by zalberico** [comment]
So far after `|ames-verb %odd` I see no new output:
```
> |ames-verb %odd
>=
ames: ~bolbex-fogdys ack crashed %exit
ames: ~bitpyx-dildus ack crashed %exit
ames: ~bolbex-fogdys ack crashed %exit
ames: ~bitpyx-dildus ack crashed %exit
ames: ~bitpyx-dildus ack crashed %exit
```

**Comment by zalberico** [comment]
Is this fixed for other people?

**Comment by yosoyubik** [comment]
This fixed it for me, do you still see it? (i'm assuming you are on ~binnec) Can you give me the stack trace you see after doing `|ames-sift ~ship` and `|ames-verb %odd %msg`?

**Comment by yosoyubik** [comment]
> Was that because I was missing %msg?

Yes, the `%odd` flag was to check if the fix was working in your case—this print [here](https://github.com/urbit/urbit/blob/develop/pkg/arvo/sys/vane/ames.hoon#L3077). If you set the `%msg` flag I'll be able to see on what bone is crashing and then compare it to what ~~`~bolbex-fogdys` and `~bitpyx-dildus` are trying to send~~ I thought these were on GCP but can't find them there. Do we know who's hosting them>

**Comment by zalberico** [comment]
They're hosted on tlon.network now - what do I need to look for on those ships?

**Comment by yosoyubik** [comment]
you can do this on your ship:

```
|ames-sift ~bolbex-fogdys
|ames-verb %odd %msg
```

the crash now will contain information about on which bone is crashing. You can go to the ~debug dashboard, search for ~bolbex-fogdys and look for that bone, that will tell us what flow is causing this.

**Comment by zalberico** [comment]
<img width="2187" alt="Screenshot 2023-06-21 at 12 22 12 PM" src="https://github.com/urbit/urbit/assets/441466/76857fc2-c56e-486c-ba7b-d23871f422db">
I'm not really sure what I should be looking for here, probably easiest to debug in a screen share

**Comment by zalberico** [comment]
Also this (for ~bitpyx-dildus)
```
/sys/vane/ames/hoon:<[4.635 3].[4.673 20]>
call: failed
bar-stack=[i=[i=//ames t=~] t=~]
[%poke %hear]
ames: ~bitpyx-dildus ack crashed %exit
[bone=176 message-num=2 meat=[%.n p=[%.n p=[ok=%.y lag=~s0]]]]
/sys/vane/ames/hoon:<[2.844 11].[2.844 54]>
[%dangling-bone ~bitpyx-dildus 176]
/sys/vane/ames/hoon:<[2.843 11].[2.844 54]>
```

<img width="2043" alt="Screenshot 2023-06-21 at 12 29 03 PM" src="https://github.com/urbit/urbit/assets/441466/6e886bae-7400-499d-bb63-d88bc5622ef8">

**Comment by yosoyubik** [comment]
@zalberico do you have the full stack-trace? I think it's crashing [here](https://github.com/urbit/urbit/blob/develop/pkg/arvo/sys/vane/ames.hoon#L3443), but want to confirm with the full output. If I'm reading this correctly, your ship is sending messages to ~bitpyx-dildus on bones 176 and 186. When you hear acks for those messages you process them, and then on `+pump-done:mu` you retrieve the bone associated with that ack... and then crashed because you don't have it anymore? The bone is not corked, otherwise you wouldn't be crashing there, and the debug output you shared clearly shows the bones are there, so there's something else going on here—probably crashing somewhere else.

**Comment by zalberico** [comment]
I don't have anything more than the output I shared (I'm not getting more output than that).

I can jump on a 1:1 though if there's a time that works (probably will save a lot of this back and forth to debug directly).

**Comment by mrdomino** [comment]
fyi I appear to have this on ~miptyr with ~wannux-laptex.
